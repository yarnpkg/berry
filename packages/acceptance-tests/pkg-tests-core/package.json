{
  "name": "pkg-tests-core",
  "private": true,
  "license": "BSD-2-Clause",
  "main": "./sources/index.ts",
  "devDependencies": {
    "@types/finalhandler": "1.1.0",
    "@types/invariant": "^2.2.30",
    "@types/klaw": "^2.0.0",
    "@types/minimatch": "^3.0.3",
    "@types/pem": "^1",
    "@types/semver": "^7.1.0",
    "@types/serve-static": "1.13.3",
    "@types/tar-fs": "^1.16.1",
    "@yarnpkg/cli": "workspace:*",
    "@yarnpkg/core": "workspace:*",
    "@yarnpkg/fslib": "workspace:*",
    "@yarnpkg/parsers": "workspace:*",
    "finalhandler": "^1.1.2",
    "invariant": "^2.2.4",
    "klaw": "^2.1.1",
    "minimatch": "^3.0.4",
    "pem": "^1.14.4",
    "pkg-tests-fixtures": "workspace:*",
    "semver": "^7.1.2",
    "serve-static": "^1.14.1",
    "super-resolve": "^1.0.0",
    "tar-fs": "^1.16.0"
  },
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com/yarnpkg/berry.git",
    "directory": "packages/acceptance-tests/pkg-tests-core"
  },
  "engines": {
    "node": ">=12 <14 || 14.2 - 14.9 || >14.10.0"
  }
}
