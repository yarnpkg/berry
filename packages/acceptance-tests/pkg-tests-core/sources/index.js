const exec = require('./utils/exec');
const fs = require('./utils/fs');
const misc = require('./utils/misc');
const tests = require('./utils/tests');

module.exports = {exec, fs, misc, tests};
