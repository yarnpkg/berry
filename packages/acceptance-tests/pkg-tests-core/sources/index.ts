import './utils/makeTemporaryEnv';
import './utils/makeTemporaryMonorepoEnv';
import * as exec  from './utils/exec';
import * as fs    from './utils/fs';
import * as misc  from './utils/misc';
import * as tests from './utils/tests';
import * as yarn  from './utils/yarn';

export {exec, fs, misc, tests, yarn};
