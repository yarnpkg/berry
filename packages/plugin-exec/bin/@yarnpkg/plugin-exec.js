/* eslint-disable */
module.exports = {
name: "@yarnpkg/plugin-exec",
factory: function (require) {
var plugin;plugin=(()=>{"use strict";var e={230:(e,t,r)=>{r.r(t),r.d(t,{execUtils:()=>a,default:()=>p});var a={};r.r(a),r.d(a,{loadGeneratorFile:()=>l,makeLocator:()=>c,makeSpec:()=>i,parseSpec:()=>s});var o=r(966),n=r(688);function s(e){const{params:t,selector:r}=o.structUtils.parseRange(e),a=n.npath.toPortablePath(r);return{parentLocator:t&&"string"==typeof t.locator?o.structUtils.parseLocator(t.locator):null,path:a}}function i({parentLocator:e,path:t,generatorHash:r,protocol:a}){const n=null!==e?{locator:o.structUtils.stringifyLocator(e)}:{},s=void 0!==r?{hash:r}:{};return o.structUtils.makeRange({protocol:a,source:t,selector:t,params:{...s,...n}})}function c(e,{parentLocator:t,path:r,generatorHash:a,protocol:n}){return o.structUtils.makeLocator(e,i({parentLocator:t,path:r,generatorHash:a,protocol:n}))}async function l(e,t,r){const{parentLocator:a,path:s}=o.structUtils.parseFileStyleRange(e,{protocol:t}),i=n.ppath.isAbsolute(s)?{packageFs:new n.CwdFS(n.PortablePath.root),prefixPath:n.PortablePath.dot,localPath:n.PortablePath.root}:await r.fetcher.fetch(a,r),c=i.localPath?{packageFs:new n.CwdFS(n.PortablePath.root),prefixPath:n.ppath.relative(n.PortablePath.root,i.localPath)}:i;i!==c&&i.releaseFs&&i.releaseFs();const l=c.packageFs,p=n.ppath.join(c.prefixPath,s);return await l.readFilePromise(p,"utf8")}const p={fetchers:[class{supports(e,t){return!!e.reference.startsWith("exec:")}getLocalPath(e,t){const{parentLocator:r,path:a}=o.structUtils.parseFileStyleRange(e.reference,{protocol:"exec:"});if(n.ppath.isAbsolute(a))return a;const s=t.fetcher.getLocalPath(r,t);return null===s?null:n.ppath.resolve(s,a)}async fetch(e,t){const r=t.checksums.get(e.locatorHash)||null,[a,n,s]=await t.cache.fetchPackageFromCache(e,r,{onHit:()=>t.report.reportCacheHit(e),onMiss:()=>t.report.reportCacheMiss(e),loader:()=>this.fetchFromDisk(e,t),skipIntegrityCheck:t.skipIntegrityCheck});return{packageFs:a,releaseFs:n,prefixPath:o.structUtils.getIdentVendorPath(e),localPath:this.getLocalPath(e,t),checksum:s}}async fetchFromDisk(e,t){const r=await l(e.reference,"exec:",t);return n.xfs.mktempPromise(async a=>{const s=n.ppath.join(a,"generator.js");return await n.xfs.writeFilePromise(s,r),n.xfs.mktempPromise(async r=>{if(await this.generatePackage(r,e,s,t),!n.xfs.existsSync(n.ppath.join(r,"build")))throw new Error("The script should have generated a build directory");return await o.tgzUtils.makeArchiveFromDirectory(n.ppath.join(r,"build"),{prefixPath:o.structUtils.getIdentVendorPath(e),compressionLevel:t.project.configuration.get("compressionLevel")})})})}async generatePackage(e,t,r,a){return await n.xfs.mktempPromise(async s=>{const i=await o.scriptUtils.makeScriptEnv({project:a.project,binFolder:s}),c=n.ppath.join(e,"runtime.js");return await n.xfs.mktempPromise(async a=>{const s=n.ppath.join(a,"buildfile.log"),l=n.xfs.createWriteStream(s),p=l,u=n.ppath.join(e,"generator"),h=n.ppath.join(e,"build");await n.xfs.mkdirPromise(u),await n.xfs.mkdirPromise(h);const f={tempDir:n.npath.fromPortablePath(u),buildDir:n.npath.fromPortablePath(h),locator:o.structUtils.stringifyLocator(t)};await n.xfs.writeFilePromise(c,`\n          // Expose 'Module' as a global variable\n          Object.defineProperty(global, 'Module', {\n            get: () => require('module'),\n            configurable: true,\n            enumerable: false,\n          });\n\n          // Expose non-hidden built-in modules as global variables\n          for (const name of Module.builtinModules.filter((name) => name !== 'module' && !name.startsWith('_'))) {\n            Object.defineProperty(global, name, {\n              get: () => require(name),\n              configurable: true,\n              enumerable: false,\n            });\n          }\n\n          // Expose the 'execEnv' global variable\n          Object.defineProperty(global, 'execEnv', {\n            value: {\n              ...${JSON.stringify(f)},\n            },\n            enumerable: true,\n          });\n        `);let d=i.NODE_OPTIONS||"";d=d.replace(/\s*--require\s+\S*\.pnp\.c?js\s*/g," ").trim(),i.NODE_OPTIONS=d,l.write(`# This file contains the result of Yarn generating a package (${o.structUtils.stringifyLocator(t)})\n`),l.write("\n");const{code:g}=await o.execUtils.pipevp(process.execPath,["--require",n.npath.fromPortablePath(c),n.npath.fromPortablePath(r),o.structUtils.stringifyIdent(t)],{cwd:e,env:i,stdin:null,stdout:l,stderr:p});if(0!==g)throw n.xfs.detachTemp(a),new Error(`Package generation failed (exit code ${g}, logs can be found here: ${s})`)})})}}],resolvers:[class{supportsDescriptor(e,t){return!!e.range.startsWith("exec:")}supportsLocator(e,t){return!!e.reference.startsWith("exec:")}shouldPersistResolution(e,t){return!1}bindDescriptor(e,t,r){return o.structUtils.bindDescriptor(e,{locator:o.structUtils.stringifyLocator(t)})}getResolutionDependencies(e,t){return[]}async getCandidates(e,t,r){if(!r.fetchOptions)throw new Error("Assertion failed: This resolver cannot be used unless a fetcher is configured");const{path:a,parentLocator:n}=s(e.range);if(null===n)throw new Error("Assertion failed: The descriptor should have been bound");const i=await l(o.structUtils.makeRange({protocol:"exec:",source:a,selector:a,params:{locator:o.structUtils.stringifyLocator(n)}}),"exec:",r.fetchOptions);return[c(e,{parentLocator:n,path:a,generatorHash:o.hashUtils.makeHash("1",i).slice(0,6),protocol:"exec:"})]}async getSatisfying(e,t,r){return null}async resolve(e,t){if(!t.fetchOptions)throw new Error("Assertion failed: This resolver cannot be used unless a fetcher is configured");const r=await t.fetchOptions.fetcher.fetch(e,t.fetchOptions),a=await o.miscUtils.releaseAfterUseAsync(async()=>await o.Manifest.find(r.prefixPath,{baseFs:r.packageFs}),r.releaseFs);return{...e,version:a.version||"0.0.0",languageName:t.project.configuration.get("defaultLanguageName"),linkType:o.LinkType.HARD,dependencies:a.dependencies,peerDependencies:a.peerDependencies,dependenciesMeta:a.dependenciesMeta,peerDependenciesMeta:a.peerDependenciesMeta,bin:a.bin}}}]}},966:e=>{e.exports=require("@yarnpkg/core")},688:e=>{e.exports=require("@yarnpkg/fslib")}},t={};function r(a){if(t[a])return t[a].exports;var o=t[a]={exports:{}};return e[a](o,o.exports,r),o.exports}return r.d=(e,t)=>{for(var a in t)r.o(t,a)&&!r.o(e,a)&&Object.defineProperty(e,a,{enumerable:!0,get:t[a]})},r.o=(e,t)=>Object.prototype.hasOwnProperty.call(e,t),r.r=e=>{"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},r(230)})();
return plugin;
}
};