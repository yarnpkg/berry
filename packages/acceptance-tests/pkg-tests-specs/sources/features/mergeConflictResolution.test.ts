import { Filename, ppath, xfs } from "@yarnpkg/fslib";
import { exec } from "pkg-tests-core";

const LOCKFILE_1_0_0 = `
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1

no-deps@*:
  version "1.0.0"
  resolved "http://localhost:64572/no-deps/-/no-deps-1.0.0.tgz#8d556c1e3e6c953ea16689c506073d5fa26b957f"
  integrity sha1-jVVsHj5slT6hZonFBgc9X6JrlX8=
`;

const LOCKFILE_2_0_0 = `
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1

no-deps@*:
  version "2.0.0"
  resolved "http://localhost:64572/no-deps/-/no-deps-2.0.0.tgz#f3b8f24d76b21b04748e6888672752b872236fa7"
  integrity sha1-87jyTXayGwR0jmiIZydSuHIjb6c=
`;

function cleanLockfile(lockfile: string) {
  lockfile = lockfile.replace(/(^ {2}version: )[0-9]+$/m, `$1X`);
  lockfile = lockfile.replace(/(checksum: ).*/g, `$1<checksum stripped>`);
  lockfile = lockfile.replace(/(>>>>>>>).*(\(commit-[0-9].0.0\))/g, `$1 0000000 $2`);

  return lockfile;
}

describe(`Features`, () => {
  describe(`Merge Conflict Resolution`, () => {
    test(
      `it should properly fix merge conflicts`,
      makeTemporaryEnv({}, async ({ path, run, source }) => {
        await exec.execGitInit({ cwd: path });

        await run(`install`);

        await xfs.writeJsonPromise(ppath.join(path, Filename.manifest), {
          dependencies: {
            [`no-deps`]: `*`,
          },
        });

        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `my-commit`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `1.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:1.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-1.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `2.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:2.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-2.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`merge`, `1.0.0`], { cwd: path });

        await expect(exec.execFile(`git`, [`merge`, `2.0.0`], { cwd: path, env: { LC_ALL: `C` } })).rejects.toThrow(
          /CONFLICT/,
        );

        const lockfile = await xfs.readFilePromise(ppath.join(path, Filename.lockfile), `utf8`);
        expect(cleanLockfile(lockfile)).toMatchSnapshot();

        await expect(run(`install`)).resolves.toMatchSnapshot();
      }),
    );

    test(
      `it should properly fix merge conflicts when old is Yarn 1 and new is Yarn 2`,
      makeTemporaryEnv({}, async ({ path, run, source }) => {
        await exec.execGitInit({ cwd: path });

        await xfs.writeJsonPromise(ppath.join(path, Filename.manifest), { dependencies: { [`no-deps`]: `*` } });
        await xfs.writeFilePromise(ppath.join(path, Filename.lockfile), LOCKFILE_1_0_0);

        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `my-commit`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `2.0.0`], { cwd: path });
        await xfs.writeFilePromise(ppath.join(path, Filename.lockfile), LOCKFILE_2_0_0);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-2.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `yarn2`], { cwd: path });
        await run(`install`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-yarn2`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`merge`, `2.0.0`], { cwd: path });

        await expect(exec.execFile(`git`, [`merge`, `yarn2`], { cwd: path, env: { LC_ALL: `C` } })).rejects.toThrow(
          /CONFLICT/,
        );

        const lockfile = await xfs.readFilePromise(ppath.join(path, Filename.lockfile), `utf8`);
        expect(cleanLockfile(lockfile)).toMatchSnapshot();

        await expect(run(`install`)).resolves.toMatchSnapshot();
      }),
    );

    test(
      `it should support fixing rebase conflicts`,
      makeTemporaryEnv({}, async ({ path, run, source }) => {
        await exec.execGitInit({ cwd: path });

        await run(`install`);
        await xfs.writeJsonPromise(ppath.join(path, Filename.manifest), { dependencies: { [`no-deps`]: `*` } });

        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `my-commit`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `1.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:1.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-1.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `2.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:2.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-2.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`merge`, `1.0.0`], { cwd: path });

        await expect(exec.execFile(`git`, [`rebase`, `2.0.0`], { cwd: path, env: { LC_ALL: `C` } })).rejects.toThrow(
          /CONFLICT/,
        );

        const lockfile = await xfs.readFilePromise(ppath.join(path, Filename.lockfile), `utf8`);
        expect(cleanLockfile(lockfile)).toMatchSnapshot();

        await expect(run(`install`)).resolves.toMatchSnapshot();
      }),
    );

    test(
      `it should support fixing cherry-pick conflicts`,
      makeTemporaryEnv({}, async ({ path, run, source }) => {
        await exec.execGitInit({ cwd: path });

        await run(`install`);
        await xfs.writeJsonPromise(ppath.join(path, Filename.manifest), { dependencies: { [`no-deps`]: `*` } });

        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `my-commit`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `1.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:1.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-1.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `2.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:2.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-2.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`merge`, `1.0.0`], { cwd: path });

        await expect(
          exec.execFile(`git`, [`cherry-pick`, `2.0.0`], { cwd: path, env: { LC_ALL: `C` } }),
        ).rejects.toThrow(/CONFLICT/);

        const lockfile = await xfs.readFilePromise(ppath.join(path, Filename.lockfile), `utf8`);
        expect(cleanLockfile(lockfile)).toMatchSnapshot();

        await expect(run(`install`)).resolves.toMatchSnapshot();
      }),
    );

    test(
      `it shouldn't re-fetch the lockfile metadata when performing simple merge conflict resolutions`,
      makeTemporaryEnv({}, async ({ path, run, source }) => {
        await exec.execGitInit({ cwd: path });

        await run(`install`);
        await xfs.writeJsonPromise(ppath.join(path, Filename.manifest), { dependencies: { [`no-deps`]: `*` } });

        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `my-commit`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `1.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:1.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-1.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`checkout`, `-b`, `2.0.0`], { cwd: path });
        await run(`set`, `resolution`, `no-deps@npm:*`, `npm:2.0.0`);
        await exec.execFile(`git`, [`add`, `-A`], { cwd: path });
        await exec.execFile(`git`, [`commit`, `-a`, `-m`, `commit-2.0.0`], { cwd: path });

        await exec.execFile(`git`, [`checkout`, `master`], { cwd: path });
        await exec.execFile(`git`, [`merge`, `1.0.0`], { cwd: path });

        await expect(exec.execFile(`git`, [`merge`, `2.0.0`], { cwd: path, env: { LC_ALL: `C` } })).rejects.toThrow(
          /CONFLICT/,
        );

        const lockfile = await xfs.readFilePromise(ppath.join(path, Filename.lockfile), `utf8`);
        expect(cleanLockfile(lockfile)).toMatchSnapshot();

        await expect(
          run(`install`, {
            enableNetwork: false,
          }),
        ).resolves.toMatchSnapshot();
      }),
    );
  });
});
