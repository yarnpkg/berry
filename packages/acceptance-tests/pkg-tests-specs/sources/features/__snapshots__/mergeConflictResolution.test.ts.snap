// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Merge Conflict Resolution it should properly fix merge conflicts 1`] = `
"# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: X
  cacheKey: 0c0

"no-deps@npm:*":
<<<<<<< HEAD
  version: 1.0.0
  resolution: "no-deps@npm:1.0.0"
  checksum: <checksum stripped>
=======
  version: 2.0.0
  resolution: "no-deps@npm:2.0.0"
  checksum: <checksum stripped>
>>>>>>> 2.0.0
  languageName: node
  linkType: hard

"root-workspace-0b6124@workspace:.":
  version: 0.0.0-use.local
  resolution: "root-workspace-0b6124@workspace:."
  dependencies:
    no-deps: "npm:*"
  languageName: unknown
  linkType: soft
"
`;

exports[`Features Merge Conflict Resolution it should properly fix merge conflicts 2`] = `
{
  "code": 0,
  "stderr": "",
  "stdout": "➤ YN0048: Automatically fixed merge conflicts 👍

➤ YN0000: · Yarn 0.0.0
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ No new packages added to the cache, but one was removed (- 0.78 KiB).
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done
",
}
`;

exports[`Features Merge Conflict Resolution it should properly fix merge conflicts when old is Yarn 1 and new is Yarn 2 1`] = `
"# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: X
  cacheKey: 0c0

<<<<<<< HEAD
no-deps@*:
  version "2.0.0"
  resolved "http://localhost:64572/no-deps/-/no-deps-2.0.0.tgz#f3b8f24d76b21b04748e6888672752b872236fa7"
  integrity sha1-87jyTXayGwR0jmiIZydSuHIjb6c=
=======
"no-deps@npm:*":
  version: 2.0.0
  resolution: "no-deps@npm:2.0.0"
  checksum: <checksum stripped>
  languageName: node
  linkType: hard

"root-workspace-0b6124@workspace:.":
  version: 0.0.0-use.local
  resolution: "root-workspace-0b6124@workspace:."
  dependencies:
    no-deps: "npm:*"
  languageName: unknown
  linkType: soft
>>>>>>> yarn2
"
`;

exports[`Features Merge Conflict Resolution it should properly fix merge conflicts when old is Yarn 1 and new is Yarn 2 2`] = `
{
  "code": 0,
  "stderr": "",
  "stdout": "➤ YN0048: Automatically fixed merge conflicts 👍

➤ YN0000: · Yarn 0.0.0
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done
",
}
`;

exports[`Features Merge Conflict Resolution it should support fixing cherry-pick conflicts 1`] = `
"# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: X
  cacheKey: 0c0

"no-deps@npm:*":
<<<<<<< HEAD
  version: 1.0.0
  resolution: "no-deps@npm:1.0.0"
  checksum: <checksum stripped>
=======
  version: 2.0.0
  resolution: "no-deps@npm:2.0.0"
  checksum: <checksum stripped>
>>>>>>> 0000000 (commit-2.0.0)
  languageName: node
  linkType: hard

"root-workspace-0b6124@workspace:.":
  version: 0.0.0-use.local
  resolution: "root-workspace-0b6124@workspace:."
  dependencies:
    no-deps: "npm:*"
  languageName: unknown
  linkType: soft
"
`;

exports[`Features Merge Conflict Resolution it should support fixing cherry-pick conflicts 2`] = `
{
  "code": 0,
  "stderr": "",
  "stdout": "➤ YN0048: Automatically fixed merge conflicts 👍

➤ YN0000: · Yarn 0.0.0
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ No new packages added to the cache, but one was removed (- 0.78 KiB).
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done
",
}
`;

exports[`Features Merge Conflict Resolution it should support fixing rebase conflicts 1`] = `
"# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: X
  cacheKey: 0c0

"no-deps@npm:*":
<<<<<<< HEAD
  version: 2.0.0
  resolution: "no-deps@npm:2.0.0"
  checksum: <checksum stripped>
=======
  version: 1.0.0
  resolution: "no-deps@npm:1.0.0"
  checksum: <checksum stripped>
>>>>>>> 0000000 (commit-1.0.0)
  languageName: node
  linkType: hard

"root-workspace-0b6124@workspace:.":
  version: 0.0.0-use.local
  resolution: "root-workspace-0b6124@workspace:."
  dependencies:
    no-deps: "npm:*"
  languageName: unknown
  linkType: soft
"
`;

exports[`Features Merge Conflict Resolution it should support fixing rebase conflicts 2`] = `
{
  "code": 0,
  "stderr": "",
  "stdout": "➤ YN0048: Automatically fixed merge conflicts 👍

➤ YN0000: · Yarn 0.0.0
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ No new packages added to the cache, but one was removed (- 0.78 KiB).
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done
",
}
`;

exports[`Features Merge Conflict Resolution it shouldn't re-fetch the lockfile metadata when performing simple merge conflict resolutions 1`] = `
"# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: X
  cacheKey: 0c0

"no-deps@npm:*":
<<<<<<< HEAD
  version: 1.0.0
  resolution: "no-deps@npm:1.0.0"
  checksum: <checksum stripped>
=======
  version: 2.0.0
  resolution: "no-deps@npm:2.0.0"
  checksum: <checksum stripped>
>>>>>>> 2.0.0
  languageName: node
  linkType: hard

"root-workspace-0b6124@workspace:.":
  version: 0.0.0-use.local
  resolution: "root-workspace-0b6124@workspace:."
  dependencies:
    no-deps: "npm:*"
  languageName: unknown
  linkType: soft
"
`;

exports[`Features Merge Conflict Resolution it shouldn't re-fetch the lockfile metadata when performing simple merge conflict resolutions 2`] = `
{
  "code": 0,
  "stderr": "",
  "stdout": "➤ YN0048: Automatically fixed merge conflicts 👍

➤ YN0000: · Yarn 0.0.0
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ No new packages added to the cache, but one was removed (- 0.78 KiB).
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done
",
}
`;
